### PR TITLE
Fixed bug #205  Test reports null BoundingRectangle on ScrollBar button

### PR DESCRIPTION
**Pull request by RobGallo**
#### PR checklist

- [ ] Run through of all [test scenarios](https://github.com/Microsoft/accessibility-insights-windows/blob/master/docs/Scenarios.md) completed?
- [ ] Does this address an existing issue? If yes, Issue# - 
- [ ] Includes UI changes?
  - [ ] Run the production version of Accessibility Insights for Windows against a version with changes.
  - [ ] Attach any screenshots / GIF's that are applicable.

Note - After the PR has been created, certain checks will be kicked off. All of these checks must pass before a merge. 

#### Describe the change

When the thumb control of a WPF scrollbar is all the way down or up, the rectangle of the PageDown or PageUp buttons may be null.

It is possible this fix could be even more specific, testing to actually see if the vertical or horizontal scroll percent of the parent scrollbar is 0 or 100%. Such a test might be overkill, but thoughts are welcome.

### CODE REVIEW

File: src/AccessibilityInsights.Rules/Library/BoundingRectangleNotNull.cs [hunk 1]
```diff
@@ -36,7 +37,10 @@ protected override Condition CreateCondition()
             //
             // 2 exceptions related with Menubar "System" and Menu item "System". 
             // These two are excluded since Windows 10 sets the bounding rectangles of these as null by default.
+            // WPF Scrollbar page up and page down  buttons may sometimes
```
Comment (DaveTryon) [inline src/AccessibilityInsights.Rules/Library/BoundingRectangleNotNull.cs]: >  [](start = 51, length = 1)

extra space between "down" and "buttons"

File: src/AccessibilityInsights.RulesTest/PropertyConditions/ElementGroupsTests.cs [hunk 2]
```diff
@@ -61,5 +61,110 @@ public void AllowSameNameAndControlType_True_ExceedsMaxLength()
...
+
+        [TestMethod]
+        public void WPFScrollbarPageUpOrPageDownButtons_NotWPF_False()
+        {
+            using (var e = new MockA11yElement())
+            using (var parent = new MockA11yElement())
+            {
+                parent.ControlTypeId = ControlType.ScrollBar;
+                e.ControlTypeId = ControlType.Button;
+                // e.Framework = "WPF";
```
Comment (DaveTryon) [inline src/AccessibilityInsights.RulesTest/PropertyConditions/ElementGroupsTests.cs]: >Framework [](start = 21, length = 9)

Remove commented line?

File: src/AccessibilityInsights.RulesTest/PropertyConditions/ElementGroupsTests.cs [hunk 3]
```diff
@@ -61,5 +61,110 @@ public void AllowSameNameAndControlType_True_ExceedsMaxLength()
...
+                e.AutomationId = "PageDown";
+                parent.Children.Add(e);
+                e.Parent = parent;
+
+                Assert.IsTrue(ElementGroups.WPFScrollBarPageUpOrPageDownButtons.Matches(e));
+            } // using
+        }
+
+        [TestMethod]
+        public void WPFScrollbarPageUpOrPageDownButtons_NotWPF_False()
```
Comment (DaveTryon) [inline src/AccessibilityInsights.RulesTest/PropertyConditions/ElementGroupsTests.cs]: >PageUpOrPageDown [](start = 32, length = 16)

Name suggests this tests both PageUp and PageDown, when it tests only PageUp

File: src/AccessibilityInsights.RulesTest/PropertyConditions/ElementGroupsTests.cs [hunk 4]
```diff
@@ -61,5 +61,110 @@ public void AllowSameNameAndControlType_True_ExceedsMaxLength()
...
+                e.AutomationId = "PageUp";
+                parent.Children.Add(e);
+                e.Parent = parent;
+
+                Assert.IsFalse(ElementGroups.WPFScrollBarPageUpOrPageDownButtons.Matches(e));
+            } // using
+        }
+
+        [TestMethod]
+        public void WPFScrollbarPageUpOrPageDownButtons_NotButton_False()
```
Comment (DaveTryon) [inline src/AccessibilityInsights.RulesTest/PropertyConditions/ElementGroupsTests.cs]: >WPFScrollbarPageUpOrPageDownButtons_NotButton_False [](start = 20, length = 51)

Name suggests this tests both PageUp and PageDown, when it tests only PageUp

File: src/AccessibilityInsights.Rules/PropertyConditions/ElementGroups.cs [hunk 5]
```diff
@@ -144,6 +145,15 @@ private static Condition CreateParentWPFDataItemCondition()
             return Condition.Create(IsParentWPFDataItem);
         }
 
+        private static Condition CreateWPFScrollBarPageUpOrPageDownButtons()
```
Comment (DaveTryon) [inline src/AccessibilityInsights.Rules/PropertyConditions/ElementGroups.cs]: This name will get pretty wordy once PageLeft and PageRight are added. Maybe CreateWPFScrollBarPageScrollButtons?

File: src/AccessibilityInsights.Rules/PropertyConditions/ElementGroups.cs [hunk 6]
```diff
@@ -28,6 +28,7 @@ static class ElementGroups
         public static Condition EmptyContainer = CreateEmptyContainerCondition();
         public static Condition ExpectedToBeFocusable = CreateExpectedToBeFocusableCondition();
         public static Condition ParentWPFDataItem = CreateParentWPFDataItemCondition();
+        public static Condition WPFScrollBarPageUpOrPageDownButtons = CreateWPFScrollBarPageUpOrPageDownButtons();
```
Comment (DaveTryon) [inline src/AccessibilityInsights.Rules/PropertyConditions/ElementGroups.cs]: >WPFScrollBarPageUpOrPageDownButtons [](start = 32, length = 35)

Maybe WPFScrollBarPageScrollButtons?

File: src/AccessibilityInsights.RulesTest/PropertyConditions/ElementGroupsTests.cs (hunk 3 above)
Comment (RobGallo) [reply src/AccessibilityInsights.RulesTest/PropertyConditions/ElementGroupsTests.cs]: No longer applicable.

**Pull request merged**